src/login.js

```diff
@@ -0,0 +1,72 @@
+const mysql = require("mysql2");
+const jwt = require("jsonwebtoken");
+const crypto = require("crypto");
+const conn = require("./db");
+
+function generateHash(salt, password) {
+  const passwordAndSalt = password + salt;
+  const hash = crypto
+    .createHash("sha256")
+    .update(passwordAndSalt)
+    .digest("hex");
+  return hash;
+}
+
+function generateToken(ID, isStaff) {
+  const twoWeeksFromNow = 2 * 7 * 24 * 60 * 60;
+  const token = jwt.sign(
+    {
+      exp: Math.floor(Date.now() / 1000) + twoWeeksFromNow,
+      ID,
+      isStaff,
+    },
+    process.env.JWT_SECRET
+  );
+  return token;
+}
+
+const login = {
+  async checkUser(req, res, next) {
```

Comment:
It is unnecessary to put the `checkUser` function into the `login` object when the only thing exported from the file is the `checkUser` function. You could just put
```js
module.exports = checkUser
```
to the end of the file, or if you insist on exporting an object then it can be
```js
module.exports = { checkUser }
```